File: server/model/VotingUserEnvironment.py
from sqlalchemy import BigInteger, Column, String, ForeignKey

from database.db import Base
from sqlalchemy.dialects.postgresql import JSONB


# ! Classe associativa (relacionamentos)
class VotingUserEnvironment(Base):
    __tablename__ = "voting_users_environments"
    __table_args__ = {"schema": "ic"}

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    voting_user_id: str = Column(
        String(45), ForeignKey("ic.voting_users.id"), nullable=False
    )
    environment_id: str = Column(
        String(45), ForeignKey("ic.environments.id"), nullable=False
    )
    votes_rcr_definition: dict = Column(JSONB)
    votes_rcr_priority: dict = Column(JSONB)

File: server-js/models/VotingUserEnvironment.js
module.exports = (sequelize, DataTypes) => {
  let VotingUserEnvironment = sequelize.define(
    "VotingUserEnvironment",
    {
      id: {
        type: DataTypes.BIGINT,
        primaryKey: true,
        autoIncrement: true,
      },
      votes_rcr_definition: {
        type: DataTypes.JSONB,
      },
      votes_rcr_priority: {
        type: DataTypes.JSONB,
      },
    },
    {
      tableName: "voting_users_environments",
      schema: "ic",
    }
  );

  return VotingUserEnvironment;
};
/*
VotingUserEnvironment.init(
  {
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true,
    },
    // !! Relacionamentos - definir dps 
    voting_user_id: {
      type: DataTypes.STRING(45),
      allowNull: false,
      references: {
        model: 'voting_users',
        key: 'id'
      }
    },
    // !! Relacionamentos - definir dps 
    environment_id: {
      type: DataTypes.STRING(45),
      allowNull: false,
      references: {
        model: 'environments',
        key: 'id'
      }
    },
    votes_rcr_definition: {
      type: DataTypes.JSONB,
    },
    votes_rcr_priority: {
      type: DataTypes.JSONB,
    },
  },
  {
    sequelize,
    modelName: "VotingUserEnvironment",
    tableName: "voting_users_environments",
    schema: "ic",
  }
);

module.exports = VotingUserEnvironment;
*/
